### PR TITLE
feat(browser history): add Chrome, Firefox, Safari, Edge, Brave, Vivaldi history search

### PR DESCRIPTION
**Pull request by crisboarna**
## Description

Added `Search Browser History` action similar to already existing `Search Browser Bookmarks` with 
- ability to enable each individual browser to be included in results
- select ordering of first browser in results
- open from any browser history in default browser
- open from any browser history in specific browser via action panel
- copy URL via action panel
- support for Chrome, Firefox, Safari, Edge, Brave, Vivaldi

## Screencast
browser-history-1
browser-history-2
browser-history-3
browser-history-4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/browser-history/src/components/actions.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { Action, ActionPanel, Detail, Icon, openCommandPreferences } from "@raycast/api";
+import { openNewChromeTab, openNewFirefoxTab, openNewSafariTab } from "../actions";
+import { HistoryEntry } from "../interfaces";
+
+export const HistoryItemAction = ({ entry: { title, url } }: { entry: HistoryEntry }) => (
+  <ActionPanel title={title}>
```
Comment (pernielsentikaer) [inline extensions/browser-history/src/components/actions.tsx:6]: I would remove the title from here since it can be very long

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* I would recommend that you use [useSQL](https://developers.raycast.com/utilities/react-hooks/usesql?q=usesql) instead of including it yourself

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/browser-history/src/components/actions.tsx (hunk 1 above)
Comment (crisboarna) [reply extensions/browser-history/src/components/actions.tsx]: It has been removed.

**Comment by crisboarna** [comment]
> I would recommend that you use [useSQL](https://developers.raycast.com/utilities/react-hooks/usesql?q=usesql) instead of including it yourself

Hi @pernielsentikaer , I have tried using the `useSQL` hook but it results in

```shell
Error: Error: in prepare, unable to open database file (14)

    at Object.current (/Users/<user>/.config/raycast/extensions/mozilla-firefox/firefox-new-tab.js:4429:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

If i run the command directly in terminal I also get the same result
CleanShot 2022-12-07 at 20 19 18@2x
whereas running it with the WASM SQL it works properly.

This is after I also had to debug my path and see that the first `sqlite3` executable picked up is from `Android SDK` which is at version `3.27.0` while the code invokes 

```javascript
const spawned = node_child_process_1.default.spawn("sqlite3", ["--json", "--readonly", databasePath, query], {
```

with `--json` flag, which seems to be available from `3.33.0`, causing a hidden issue for me.

In summary, could I please continue using the embedded WASM SQL module please ?

**Comment by crisboarna** [comment]
Hi @pernielsentikaer ,

Did a refactor to use `useSQL` and the same issue happens with it. Seems like there is a bug in `useSQL` itself when trying to access Firefox `places.sqlite` file. Same issue happens as in above comment if i try `sqlite3` directly.

CleanShot 2022-12-10 at 19 59 02@2x
CleanShot 2022-12-10 at 19 59 49@2x

Could this extension use WASM module as well in the meantime as other extensions are doing please ?

**Comment by crisboarna** [comment]
Raised #3853 issue for tracking as well with minimum reproducible code snippet.

**Comment by crisboarna** [comment]
Updated PR to add Edge, Brave and Vivaldi browsers history search.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥
